content/400-reference/200-api-reference/100-prisma-schema-reference.mdx

```diff
@@ -2936,6 +2936,48 @@ model User {
 </tab>
 </TabbedContent>
 
+### <inlinecode>sequence()</inlinecode>
+
+<Admonition type="warning">
+
+**Only supported by CockroachDB** <br />
+The sequence function is for now only supported by [CockroachDB connector](/concepts/database-connectors/cockroachdb).
+
+</Admonition>
+
+Create a sequence of integers in the underlying database and assign the incremented values to the ID values of the created records based on the sequence.
+
+#### Optional arguments
+
+| Argument    | Example                                               | Description                                                                                                                                                                                                                                           |
+|-------------|-------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `virtual`   | `@default(sequence(virtual))`                         | Virtual sequences are sequences that do not generate monotonically increasing values and instead produce values like those generated by the built-in function `unique_rowid()`. They are intended for use in combination with `SERIAL`-typed columns. |
+| `cache`     | `@default(sequence(cache: 20))`                       | The number of sequence values to cache in memory for reuse in the session. A cache size of `1` means that there is no cache, and cache sizes of less than `1` are not valid.                                                                          |
```

Comment:
Is 1 the default here @tomhoule?

Reply:
Yes, that's what cockroach has in its catalog if you create a sequence without specifying the cache parameter.

Reply:
I assume it should be invisible to the user, but it may have performance implications.

Reply:
https://www.cockroachlabs.com/docs/stable/create-sequence.html#cache-sequence-values-in-memory

> For improved performance, use the CACHE keyword to cache sequence values in memory.

Reply:
And here is the response: https://glennfawcett.wpcomstaging.com/2021/08/25/seqcache/